### PR TITLE
Feature/cart-page

### PR DESCRIPTION
**Pull request by rishupatel**
cart_desktop

cart_view_mobile

apply_coupon

apply_coupon_mobile

redeem

redeem_coupon_mobile

empty_cart_deskto

empty_cart_mobile

### CODE REVIEW

File: src/components/CartDrawer/CartDrawer.tsx [hunk 1]
```diff
@@ -1,39 +1,48 @@
...
     const handleCheckOutClick = (): void => {
         dispatch(closeCartDrawer());
-        if (user) {
-            navigate(ROUTES.CART);
-        } else {
-            navigate(ROUTES.MOBILE_LOGIN);
-        }
+        navigate(ROUTES.CART);
+
+        // FIXME: ix routing logic
```
Comment (prateekoli9) [inline src/components/CartDrawer/CartDrawer.tsx:39]: Can this be fixed now?

File: src/components/CartPreview/CartPreview.component.tsx [hunk 2]
```diff
@@ -8,18 +8,19 @@ import { AiOutlineClose } from 'react-icons/ai';
...
 import { ROUTES } from 'helpers/routes.helper';
-import { useNavigate } from 'react-router-dom';
+import { useLocation, useNavigate } from 'react-router-dom';
 
 const CartPreview = (): JSX.Element => {
     const cartProducts = useSelector((state: Store) => state.cart.products);
     const isCartPreviewVisible: boolean = useSelector((state: Store) => state.app.isCartPreviewVisible);
     const cartErrorMessage: string = useSelector((state: Store) => state.cart.errorMessage);
-    const { user } = useSelector((state: Store) => state.users);
+    // const { user } = useSelector((state: Store) => state.users);
```
Comment (prateekoli9) [inline src/components/CartPreview/CartPreview.component.tsx:17]: Remove commented code

File: src/components/CartPreview/CartPreview.component.tsx [hunk 3]
```diff
@@ -30,24 +31,40 @@ const CartPreview = (): JSX.Element => {
...
-        if (user) {
-            navigate(ROUTES.CART);
-        } else {
-            navigate(ROUTES.MOBILE_LOGIN);
-        }
+
+        // FIXME: fix the routing
+        navigate(ROUTES.CART);
+
+        // if (user) {
```
Comment (prateekoli9) [inline src/components/CartPreview/CartPreview.component.tsx:38]: Can this be fixed now?

**Review by prateekoli9 — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 4]
```diff
@@ -51,6 +55,27 @@ function App(): JSX.Element {
                 {/* bckdrop for side navigation */}
                 {isSideNavOpen ? <Backdrop /> : null}
                 <TopScroll />
+
+                <Dialog
```
Comment (prateekoli9) [inline src/App.tsx:59]: Make sure this doesn't affect the existing functionality

File: src/components/CartDrawer/CartDrawer.tsx (hunk 1 above)
Comment (rishupatel) [reply src/components/CartDrawer/CartDrawer.tsx:39]: done

File: src/components/CartPreview/CartPreview.component.tsx (hunk 2 above)
Comment (rishupatel) [reply src/components/CartPreview/CartPreview.component.tsx:17]: fixed

**Pull request merged**